### ISSUE TITLE
[Feature] Playwright codegen plugin system 

### ISSUE THREAD

**Issue by viktor-silakov** [microsoft/playwright#28474]
Is it possible to create a plugin system so that you can add your own features? Specifically, I need to add a couple of buttons to this panel with an overlay to integrate with my visual testing tool:

<img width="244" alt="image" src="https://github.com/microsoft/playwright/assets/23633060/3c90671d-9386-4544-b52c-0b8ebbd43986">

I studied the sources and managed to do this in the playwright fork by changing several files:

<img width="311" alt="image" src="https://github.com/microsoft/playwright/assets/23633060/dedcbcae-0ced-40eb-ba12-c8afb80f7204">

But this solution is not very good because it is not extensible and in the future when updating I will not be able to use it.

What are the prospects for creating plugin features?

**Comment by TroubleZjb** [comment]
Same situation here. My team is seeking more sustainability of testing maintenance, as one way, we plan to use a locator function with fallback logic (for example, nth-child selector -> class selector -> by position -> byText).

For now, one solution is to fork playwright and patch the recorder code, whitch is not good for its stability over long term.

A plugin system maybe a better way to do that

**Comment by jan-molak** [comment]
It would be great if Playwright Codegen allowed external frameworks building on top of Playwright, like [Serenity/JS](https://serenity-js.org/handbook/test-runners/playwright-test/), to generate code using framework-specific syntax.

This should be possible to accomplish if external frameworks were able to register their [custom language generators](https://github.com/microsoft/playwright/blob/release-1.43/packages/playwright-core/src/server/recorder.ts#L435-L447).

I'd be happy to collaborate on this.

**Comment by kotosha-real** [comment]
My pain over the lack of a plugin system is based on two issues I need to address in my project:
1. Ability to take screenshots during codgen
2. Adding some stability improvements like `waitForEvent()` after the page is opened

This issue is now 6 months old, can you please tell me if this is something you will be working on in the near (2-3 months) future? @mxschmitt tagging you as you placed the badge on my issue above

**Comment by jiwon-mun** [comment]
absolutely necessary 🥲🥲

**Comment by bhreinb** [comment]
Yeah I would be like to have the `plugin system` so to add support for other languages/frameworks that have an integration with `SerenityJs` but isn't necessarily supported by Playwright Codegen out of the box.

**Comment by autonomobil** [comment]
any updates here?

**Comment by UttamAaseri** [comment]
any plan to support this
